### PR TITLE
Feature/rocksdb validate sst

### PR DESCRIPTION
**Pull request by cpjulia**
Documentation for https://github.com/arangodb/arangodb/pull/16693

Depends on #1090

### CODE REVIEW

File: 3.10/programs-arangod-rocksdb.md [hunk 1]
```diff
@@ -240,6 +240,17 @@ Approximate size of user data (in bytes) packed per block for uncompressed data.
...
+### Validation
+
+`--rocksdb.verify-sst`
+
+This flag specifies whether or not to validate the sst files present in the 
+database directory provided on startup. If set to true, during startup, all the
+sst files in `engine-rocksdb` of the database directory will be checked for 
+validation and then the execution will be finished, so this parameter treats 
+the server startup as a means to execute the file validation as a tool and then
+stop execution. Default: false.
```
Comment (jsteemann) [inline 3.10/programs-arangod-rocksdb.md:252]:
```suggestion
sst files in `engine-rocksdb` of the database directory will be checked for 
potential corruption and errors. The server process will stop after the check
and return an exit code of `0` if the validation was successful, or a non-zero
exit code if there is an error in any of the SST files. Default: false.
```

File: 3.10/programs-arangod-rocksdb.md [hunk 2]
```diff
@@ -240,6 +240,17 @@ Approximate size of user data (in bytes) packed per block for uncompressed data.
 If true, keeps a pool of log files around for recycling them. The default
 value is false.
 
+### Validation
+
+`--rocksdb.verify-sst`
+
+This flag specifies whether or not to validate the sst files present in the
```
Comment (jsteemann) [inline 3.10/programs-arangod-rocksdb.md:247]:
```suggestion
This flag specifies whether or not to validate the SST files present in the 
```

File: 3.10/programs-arangod-rocksdb.md [hunk 3]
```diff
@@ -240,6 +240,17 @@ Approximate size of user data (in bytes) packed per block for uncompressed data.
 If true, keeps a pool of log files around for recycling them. The default
 value is false.
 
+### Validation
+
+`--rocksdb.verify-sst`
+
+This flag specifies whether or not to validate the sst files present in the 
+database directory provided on startup. If set to true, during startup, all the
+sst files in `engine-rocksdb` of the database directory will be checked for
```
Comment (jsteemann) [inline 3.10/programs-arangod-rocksdb.md:249]:
```suggestion
SST files in `engine-rocksdb` of the database directory will be checked for 
```

File: 3.10/release-notes-new-features310.md [hunk 4]
```diff
@@ -357,6 +357,11 @@ See [Responding to Liveliness Probes](http/general.html#responding-to-liveliness
 
 ### RocksDB startup options
 
+There's a new startup option `--rocksdb.verify-sst` for validating the sst files currently
+contained in the database directory on startup. If set to true, on startup, all sst files
+in the `engine-rocksdb` directory of the database will be validated, then the program will
+finish execution. The default value is `false`.
```
Comment (jsteemann) [inline 3.10/release-notes-new-features310.md:363]:
```suggestion
There's a new startup option `--rocksdb.verify-sst` for validating the SST files currently
contained in the database directory on startup. If set to `true`, on startup, all SST files
in the `engine-rocksdb` directory of the database will be validated, then the process will
finish execution. The default value is `false`.
```

File: 3.10/release-notes-new-features310.md [hunk 5]
```diff
@@ -357,6 +357,11 @@ See [Responding to Liveliness Probes](http/general.html#responding-to-liveliness
 
 ### RocksDB startup options
 
+There's a new startup option `--rocksdb.verify-sst` for validating the SST files currently
+contained in the database directory on startup. If set to `true`, on startup, all SST files
+in the `engine-rocksdb` directory of the database will be validated, then the process will
+finish execution. The default value is `false`.
```
Comment (Simran-B) [inline 3.10/release-notes-new-features310.md:363]: database directory path? of the database directory?
```suggestion
in the `engine-rocksdb` folder in the database directory are validated, then the process
finishes execution. The default value is `false`.
```

File: 3.10/release-notes-new-features310.md [hunk 6]
```diff
@@ -357,6 +357,11 @@ See [Responding to Liveliness Probes](http/general.html#responding-to-liveliness
 
 ### RocksDB startup options
 
+There's a new startup option `--rocksdb.verify-sst` for validating the SST files currently
```
Comment (Simran-B) [inline 3.10/release-notes-new-features310.md:360]:
```suggestion
The new `--rocksdb.verify-sst` startup option lets you validate the SST files currently
```

**Review by Simran-B — COMMENTED** [review]
@ansoboleva For the new option to show up in the startup option table, the 3.10 examples need to be re-generated.

**Comment by ansoboleva** [comment]
@jsteemann @cpjulia The upstream PR is for the devel branch, without a backport to 3.10. But in the docs 3.10 release notes are updated in this PR. Is this new option for devel (3.11) or 3.10?

**Review by Simran-B — APPROVED** [review]

**Pull request merged**